go/teams/storage.go

```diff
@@ -16,47 +16,106 @@ import (
 
 // Store TeamData's on memory and disk. Threadsafe.
 type Storage struct {
+	*storageGeneric
+}
+
+// Increment to invalidate the disk cache.
+const diskStorageVersion = 10
+const MemCacheLRUSize = 200
+
+type DiskStorageItem struct {
+	Version int                `codec:"V"`
+	State   *keybase1.TeamData `codec:"S"`
+}
+
+var _ teamDataGeneric = (*keybase1.TeamData)(nil)
+var _ diskItemGeneric = (*DiskStorageItem)(nil)
+
+func (d *DiskStorageItem) version() int {
+	return d.Version
+}
+func (d *DiskStorageItem) value() teamDataGeneric {
+	return d.State
+}
+func (d *DiskStorageItem) setVersion(i int) {
+	d.Version = i
+}
+func (d *DiskStorageItem) setValue(v teamDataGeneric) error {
+	typed, ok := v.(*keybase1.TeamData)
+	if !ok {
+		return fmt.Errorf("teams.Storage#Put: Bad object for setValue; got type %T", v)
+	}
+	d.State = typed
+	return nil
+}
+
+func NewStorage(g *libkb.GlobalContext) *Storage {
```

Comment:
add a comment acknowledging past mistakes so the next person doesn't spend time investigating
// DBChatInbox is used here. Oops. Next time the teams cache is busted we can switch it.